src/main/java/edu/wpi/grip/ui/SocketHandleView.java

```diff
@@ -88,29 +85,20 @@ public SocketHandleView(EventBus eventBus, SocketControlView controlView) {
         // When the user drops the connection onto another socket, add a new connection.
         this.setOnDragDropped(dragEvent -> {
             draggingSocket.ifPresent(other -> {
-                Socket inputSocket, outputSocket;
-
-                // Check which socket was the input and which was the output.  The user can create a connection by
-                // dragging either into the other.
-                if (other.isInputSocket()) {
-                    inputSocket = other.getSocket();
-                    outputSocket = this.controlView.getSocket();
-                } else {
-                    inputSocket = this.controlView.getSocket();
-                    outputSocket = other.getSocket();
+                try {
+                    Connection connection = this.controlView.getSocket().connect(other.getSocket());
```

Comment:
This hides the implementation details from the UI. The UI doesn't need to know how a socket connects just that it does connect.